Automattic/woocommerce-payments
# BNPL - Integrate BNPL offer messaging for variations in variable products

Fixes #6591 

#### Changes proposed in this Pull Request

Choosing different product variations in a variable product should update the relevant offer within BNPL Payment messaging on the product page. Until now it worked fine in simple products, the PR makes it work correctly in variable products too. ( Affirm and Afterpay offers rendered via Stripe Payment Messaging Element ) . 

#### Screen Capture

<details><summary>Before</summary>

![bnpl-before](https://github.com/Automattic/woocommerce-payments/assets/4162931/37dd6a12-cd40-4b80-a5ab-16735d75867f)

</details> 

<details><summary>After</summary>

![bnpl-variable-1](https://github.com/Automattic/woocommerce-payments/assets/4162931/4643eb10-b6ca-43d7-be90-51fa3a06ad8d)

</details> 

#### Testing instructions

* Make sure the test site has Affirm and Afterpay enabled. 
* Browse to a variable product - Typically `Hoodie` on a test site
* The site should show an offer based on the default variant - `Without logo` 
* Change the variant to with logo - The BNPL offer should update. ( Check the `After` screen capture above ) 
* Try this with various quantities too - Observe if the offer message changes accordingly
* Click the `clear` button beside the variation dropdown, the message should go blank, until a valid variation is selected. 

-------------------

- [x] Tested on mobile (or does not apply) - Tested on mobile simulation via browser tools. 

**Post merge**
